logprep/ng/abc/processor.py

```diff
@@ -208,29 +152,29 @@ def _process_rule_tree_once(tree: RuleTree, event: dict):
         else:
             _process_rule_tree_once(tree, event)
 
-    def _apply_rules_wrapper(self, event: dict, rule: "Rule"):
+    def _apply_rules_wrapper(self, event: dict, rule: "Rule") -> None:
         try:
             self._apply_rules(event, rule)
         except ProcessingWarning as error:
             self._handle_warning_error(event, rule, error)
         except ProcessingCriticalError as error:
-            if self.result is None:
+            if self._event is None:
                 raise error
-            self.result.errors.append(error)  # is needed to prevent wrapping it in itself
+            self._event.errors.append(error)  # is needed to prevent wrapping it in itself
             event.clear()
         except Exception as error:  # pylint: disable=broad-except
-            if self.result is None:
+            if self._event is None:
```

Comment:
```suggestion
            if self._event is None or not self._event.data:
```

Maybe check also, that events data is not empty?

Reply:
according to the existing implementation this is not needed or perhaps not the intended behavior